diff -Nur gg-0.2.3/src/callbacks.c gg-20020705/src/callbacks.c
--- gg-0.2.3/src/callbacks.c	Sat Jun 29 01:03:06 2002
+++ gg-20020705/src/callbacks.c	Mon Jul  1 22:04:10 2002
@@ -1,4 +1,4 @@
-/* $Id$ */
+/* $Id$ */
 
 /*
  * (C) Copyright 2001-2002 Igor Popik. Released under GPL license.
@@ -544,8 +544,8 @@
     gchar _uin[25];
     
     label = lookup_widget(window, "quick_info_label");
+    g_return_if_fail(label == NULL);
     gtk_widget_show(label);
-    g_return_if_fail(label != NULL);
 
     if (!k->uin) 
 	_uin[0] = 0;
