diff -urN gg-0.2.4.orig/src/callbacks.c gg-0.2.4/src/callbacks.c
--- gg-0.2.4.orig/src/callbacks.c	Sun Jul  7 14:30:35 2002
+++ gg-0.2.4/src/callbacks.c	Wed Jul 17 11:34:02 2002
@@ -759,9 +759,10 @@
 void on_search_unknown_clicked(GtkWidget * button, GGSession * sesja)
 {
 	GGSearch *s;
-	
-	gtk_widget_destroy(wyniki_szukaj);
-	
+
+	if (GTK_IS_WIDGET(wyniki_szukaj) && wyniki_szukaj != NULL)
+		gtk_widget_destroy(wyniki_szukaj);
+
 	s = g_new0(GGSearch, 1);
 	s->mode = 3;
 	s->uin = sesja->uin;

